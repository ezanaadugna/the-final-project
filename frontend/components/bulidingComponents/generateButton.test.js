import React from "react";
import { render } from "@testing-library/react-native";
import GenerateButtonComponent from "./GenerateButton";

// Mock the module
jest.mock("@testing-library/react-native", () => {
  return {
    getByType: jest.fn(),
  };
});

// Stub implementation
import { getByType } from "@testing-library/react-native";
getByType.mockImplementation(() => {
  return { props: { onPress: jest.fn() } };
});

describe("GenerateButtonComponent", () => {
  it("passes onPress to TouchableOpacity", () => {
    const onPress = jest.fn();

    render(<GenerateButtonComponent onPress={onPress} />);

    const button = getByType();

    button.props.onPress();

    expect(onPress).toHaveBeenCalledTimes(1);
  });
});
