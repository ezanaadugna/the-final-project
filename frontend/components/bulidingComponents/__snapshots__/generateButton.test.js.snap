// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderRadius": 10,
      "elevation": 3,
      "height": "10%",
      "justifyContent": "center",
      "marginTop": 25,
      "opacity": 1,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 4,
      "width": "50%",
    }
  }
>
  <Text
    style={
      {
        "color": "black",
        "fontFamily": "Helvetica",
        "fontSize": 17,
      }
    }
  >
    GENERATE LINES
  </Text>
</View>
`;
